contrib/rabbitmq/pkg/adapter/adapter.go

```
@@ -0,0 +1,311 @@
+
+	err := (*ch).ExchangeDeclare(
+		exchangeConfig.Name,
+		exchangeConfig.TypeOf,
+		wabbit.Option{
+			"durable":  exchangeConfig.Durable,
+			"delete":   exchangeConfig.AutoDeleted,
+			"internal": exchangeConfig.Internal,
+			"noWait":   exchangeConfig.NoWait,
+		},)
+	if err != nil {
+		logger.Error(err)
+		return nil, err
+	}
+
+	queue, err := (*ch).QueueDeclare(
+		a.QueueConfig.Name,
+		wabbit.Option{
+			"durable":   a.QueueConfig.Durable,
+			"delete":    a.QueueConfig.DeleteWhenUnused,
+			"exclusive": a.QueueConfig.Exclusive,
+			"noWait":    a.QueueConfig.NoWait,
+		},)
+	if err != nil {
+		logger.Error(err)
+		return nil, err
+	}
+
+	if a.ExchangeConfig.TypeOf != "fanout" {
+		routingKeys := strings.Split(a.QueueConfig.RoutingKey, ",")
+
+		for _, routingKey := range routingKeys {
+			err = (*ch).QueueBind(
+				queue.Name(),
+				routingKey,
+				exchangeConfig.Name,
+				wabbit.Option{
+					"noWait": a.QueueConfig.NoWait,
+				},)
+			if err != nil {
+				logger.Error(err)
+				return nil, err
+			}
+		}
+	} else {
+		err = (*ch).QueueBind(
+			queue.Name(),
+			"",
+			exchangeConfig.Name,
+			wabbit.Option{
+				"noWait": a.QueueConfig.NoWait,
+			})
+		if err != nil {
+			logger.Error(err)
+			return nil, err
+		}
+	}
+
+	return &queue, nil
+}
+
+func (a *Adapter) ConsumeMessages(channel *wabbit.Channel,
+	queue *wabbit.Queue, logger *zap.SugaredLogger) (<-chan wabbit.Delivery, error) {
+	msgs, err := (*channel).Consume(
+		(*queue).Name(),
+		"",
+		wabbit.Option{
+			"noAck":     false,
+			"exclusive": a.QueueConfig.Exclusive,
+			"noLocal":   false,
+			"noWait":    a.QueueConfig.NoWait,
+		},)
+
+	if err != nil {
+		logger.Error(err)
+	}
+	return msgs, err
+}
+
+func (a *Adapter) PollForMessages(ctx context.Context, channel *wabbit.Channel,
+	queue *wabbit.Queue, stopCh <-chan struct{}) error {
+	logger := logging.FromContext(ctx)
+
+	msgs, _ := a.ConsumeMessages(channel, queue, logger)
```

Why ignore the error?